docs/notebooks/annoytutorial.ipynb

```diff
@@ -0,0 +1,356 @@
+      "Wall time: 150 ms\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "[('the', 1.0),\n",
+       " ('in', 0.999877393245697),\n",
+       " ('two', 0.9998754858970642),\n",
+       " ('its', 0.9998745322227478),\n",
+       " ('world', 0.9998732209205627)]"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "#Traditional implementation:\n",
+    "model.most_similar([vector], topn=5)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "('the', 1.0)\n",
+      "('in', 0.9921713043004274)\n",
+      "('two', 0.9921106304973364)\n",
+      "('its', 0.9920776467770338)\n",
+      "('world', 0.9920363733544946)\n",
+      "CPU times: user 0 ns, sys: 0 ns, total: 0 ns\n",
+      "Wall time: 678 µs\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "#Annoy implementation:\n",
+    "neighbors = annoy_index.most_similar(vector, 5)\n",
+    "for neighbor in neighbors:\n",
+    "    print neighbor"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A similarity query using Annoy is significantly faster then using the traditional brute force method\n",
+    ">**Note**: Initialization time for the annoy indexer was not included in the times. The optimal knn algorithm for you to use will depend on how many queries you need to make and the size of the corpus. If you are making very few similarity queries, the time taken to initialize the annoy indexer will be longer then the time it would take the brute force method to retrieve results. If you are making many queries however, the time it takes to initialize the annoy indexer will be made up for by the incredibly fast retrieval times for queries once the indexer has been initialized"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## What is Annoy?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "[Annoy](https://github.com/spotify/annoy \"Link to annoy repo\") is an open source library to search for points in space that are close to a given query point. It also creates large read-only file-based data structures that are mmapped into memory so that many processes may share the same data. For our purpose, it is used to find similarity between words or documents in a vector space. [See the tutorial on similarity queries for more information on them](https://github.com/RaRe-Technologies/gensim/blob/develop/docs/notebooks/Similarity_Queries.ipynb).\n",
+    "\n",
+    "[There are benefits](https://github.com/spotify/annoy#background) of using annoy over the pre-existing method of making similarity queries through brute force in gensim."
```

Comment:
Gensim indexes are even easier to memory-map, so this is no benefit over gensim (comment incorrect).